### PR TITLE
Support cuda12 jar's release [skip CI]

### PR DESCRIPTION
**Pull request by NvTimLiu**
Deploy cuda12 classifer jar to OSS staging repo for release

Signed-off-by: Tim Liu <timl@nvidia.com>

### CODE REVIEW

**Comment by NvTimLiu** [comment]
build

**Comment by NvTimLiu** [comment]
CI job  successfully uploaded cuda12 jar to OSS staging repo

File: jenkins/deploy.sh [hunk 1]
```diff
@@ -40,24 +40,37 @@ DIST_PL=${DIST_PL:-"dist"}
...
+DEPLOY_TYPES=''
+DEPLOY_FILES=''
+IFS=',' read -a CUDA_CLASSIFIERS_ARR <<< "$CUDA_CLASSIFIERS"
+for classifier in "${CUDA_CLASSIFIERS_ARR[@]}"; do
+    DEPLOY_TYPES="${DEPLOY_TYPES},jar"
+    DEPLOY_FILES="${DEPLOY_FILES},${FPATH}-${classifier}.jar"
+done
+# Remove the first char ','
+DEPLOY_TYPES=${DEPLOY_TYPES#*,}
+DEPLOY_FILES=${DEPLOY_FILES#*,}
```
Comment (jlowe) [inline jenkins/deploy.sh:67]: Maybe I'm missing something, but parsing the classifiers into an array and then trying to walk the array seems more complicated than it needs to be.  Instead I think we can simply use CUDA_CLASSIFIERS directly, preserving the commas and replacing what we need.  For example, I think the following is equivalent and doesn't require fixups at the end:
```suggestion
DEPLOY_TYPES=$(echo $CUDA_CLASSIFIERS | sed -e 's;[^,]*;jar;g')
DEPLOY_FILES=$(echo $CUDA_CLASSIFIERS | sed -e "s;\([^,]*\);${FPATH}-\1.jar;g")
```

Comment (NvTimLiu) [reply jenkins/deploy.sh]: Good suggestion, thanks a lot!

File: jenkins/deploy.sh [hunk 2]
```diff
@@ -90,9 +103,9 @@ $DEPLOY_CMD -Dfile=./pom.xml -DpomFile=./pom.xml
...
 $DEPLOY_CMD -DpomFile=$POM_FILE \
-            -Dfile=$FPATH.jar \
+            -Dfile=$FPATH-$DEFAULT_CUDA_CLASSIFIER.jar \
             -Dsources=$FPATH-sources.jar \
             -Djavadoc=$FPATH-javadoc.jar \
-            -Dfiles=$FPATH-$CUDA_CLASSIFIER.jar \
-            -Dtypes=jar \
-            -Dclassifiers=$CUDA_CLASSIFIER
+            -Dfiles=$DEPLOY_FILES \
+            -Dtypes=$DEPLOY_TYPES \
```
[the remark this answers is not in the document]
Comment (NvTimLiu) [reply jenkins/deploy.sh:105]: Looks like we are  "double-deploying"   DEFAULT_CUDA_CLASSIFIER jar here, but `-Dfile=` as the main jar is a must vs `-Dfiles` as the classifier jars. 

So if user only set CUDA_CLASSIFIERS=cuda12, the the default DEFAULT_CUDA_CLASSIFIER=cuda11 jar is uploaded as the main jar, and only have cuda12 classifier jar, but no cuda11 jar file be deployed

**Review by tgravescs — APPROVED** [review]

**Comment by NvTimLiu** [comment]
build

**Pull request merged**